#2526 Build a Static Program Entity Page

daniel-cy-lu opened the pull request:
# Description of changes

Create a Program Entity Page UI, with a directory of `root/program`. The design is consistent with the rest of the page.

## Type of Change

- [ ] Bug
- [ ] Dependency updates
- [x] Feature
- [ ] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

Screenshot of Regular View
<img width="1434" alt="Screenshot 2023-02-10 at 3 59 08 PM" src="https://user-images.githubusercontent.com/79996555/218196614-86dca6b9-a5ae-4bec-a59e-73748240911d.png">

Screenshot of Shrunk Width:
<img width="849" alt="Screenshot 2023-02-10 at 3 59 36 PM" src="https://user-images.githubusercontent.com/79996555/218196447-329b4dfc-d74e-4510-a294-3b2caea6e4e0.png">

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/ProgramSummaryLinkContatiner/ProgramSummaryLink.tsx:
```diff
@@ -0,0 +1,76 @@
+  totalNum,
+  subtitle,
+}: {
+  circleFill: any;
```
For the circleFill, iconFill props, I tried assigning `ThemeColorNames` which are the type of `fill` prop of `CircleContainer` and `Icon`. It didn't work. Would someone know the reason why?

daniel-cy-lu replied:
Thanks for the comment, Dan. Let me take a look!

samrichca replied:
those other components are using `keyof ThemeColorNames` which creates a string literal type, i.e. `success | warning | error | grey ...`

samrichca replied:
I didn't see Dan's comment because I had this tab open for a long time. good idea @demariadaniel

daniel-cy-lu replied:
I updated `circleFill` and `iconFill's` type to` keyof ThemeColorNames` and they stopped screaming!
<img width="490" alt="Screenshot 2023-02-13 at 12 09 22 PM" src="https://user-images.githubusercontent.com/79996555/218525179-0b24cec2-e670-41bb-9354-0e97a79877d3.png">

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/ProgramDonorAndFileCountsTable/index.tsx:
```diff
@@ -0,0 +1,61 @@
+  );
+};
+
+export default ProgramDonorAndFileCountsTable;
```
This Component produces the two tables at the bottom of the page. We are using table with title.

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/ProgramSummaryLinkContatiner/ProgramSummaryLink.tsx:
```diff
@@ -0,0 +1,76 @@
+  );
+};
+
+export default ProgramSummaryLink;
```
The links are hidden for this UI ticket. I call them link. This Component produces the three icons and their corresponding links at the top right part of the page.

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx:
```diff
@@ -0,0 +1,62 @@
+  );
+};
+
+export default ProgramSummaryLinkContainer;
```
This Component is the container of the icon link at the top right part of the page.

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/ProgramSummaryTable/index.tsx:
```diff
@@ -0,0 +1,44 @@
+  );
+};
+
+export default ProgramSummaryTable;
```
This Component is the table at the upper part of the page. It is using a simple table because titles are not required.

daniel-cy-lu commented on components/pages/program/ProgramCardsLayout/index.tsx:
```diff
@@ -0,0 +1,128 @@
+  );
+};
+
+export default ProgramCardsLayout;
```
This Component is the container of body of the page. It holds the three major parts of the body of the page.

samrichca commented on pages/program/index.tsx:
```diff
@@ -0,0 +1,27 @@
+ */
+
+import { createPage } from 'global/utils/pages';
+import Program from 'components/pages/program';
```
this should be changed to `import ProgramEntity from 'components/pages/ProgramEntity'` for consistency with donor & file entities

samrichca commented on pages/program/index.tsx:
```diff
@@ -0,0 +1,27 @@
+export default createPage<{ egoJwt: string }>({
+  isPublic: true,
+})(() => {
+  return <Program />;
```
change to `<ProgramEntity />`

daniel-cy-lu replied:
edit files, and component name to program-entity and `ProgramEntity`

daniel-cy-lu commented:
@samrichca spelling mistake on `Strategies` updated.

samrichca commented on components/pages/program/ProgramCardsLayout/index.tsx:
```diff
@@ -0,0 +1,128 @@
+  ];
+
+  const donorAndFileCountsByExperimentalStrategy = [
+    { Stragegies: 'WXS', Donors: 245, Files: 245 },
```
strategies spelling again ;)

also please leave a comment indicating these are mock data

daniel-cy-lu replied:
It's good now after refreshing the page. I think it was still showing the wrong spelling because I just updated the file name.

samrichca commented:
looks good now, thanks for the quick changes 💯

samrichca commented on components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/ProgramSummaryLink.tsx:
```diff
@@ -0,0 +1,77 @@
+  totalNum,
+  subtitle,
+}: {
+  circleFill: any;
```
change to `keyof ThemeColorNames`

we should add a proper type to uikit, but keyof is OK for now

daniel-cy-lu replied:
updated the type!
<img width="490" alt="Screenshot 2023-02-13 at 12 09 22 PM" src="https://user-images.githubusercontent.com/79996555/218525539-36dac609-da16-4c9a-8023-e8c80d2255bf.png">

ciaranschutte commented on components/pages/program-entity/ProgramCardsLayout/ProgramDonorAndFileCountsTable/index.tsx:
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
```
2023?

daniel-cy-lu replied:
oh why didn't I thought of that!? :)

ciaranschutte replied:
copy / paste .... I only caught it because I was pasting 2022 30mins ago ;p

daniel-cy-lu replied:
I changed the year, waiting for Sam if she has a new script!

ciaranschutte replied:
it's only a couple of files.. I may have gotten over zealous about copyright script use case :p

ciaranschutte commented:
2023 for all copyright. i thnk @samrichca had a script?

ciaranschutte commented on components/pages/program-entity/index.tsx:
```diff
@@ -0,0 +1,75 @@
+import ProgramCardsLayout from './ProgramCardsLayout';
+
+const ProgramEntity = () => {
+  const loading = false;
```
`loading` isn't doing anything so isn't needed. similarly down when you use it as a conditional can just be shortened as `loading` will always be false, so <DnaLoader> is never going to render

daniel-cy-lu replied:
I think I left it there to remind whoever that is working on the API that there is a loading here. I got the idea from other pages like donor-entity and file-entity. I should and will leave a comment here for its purpose.

ciaranschutte replied:
Good future thinking!

Those pages are currently "doing stuff" that has a loading state.
There may not be any loading at all on this page ever, or loading may be constrained to child components, then there's unreachable code with extra data of an unused component. (chances are there will be, trying to illustrate overall not to add unneeded code)

It's the responsibility of the developer who implements the improvements to this page to add it :)

daniel-cy-lu replied:
AHH! I got the point! Thank Ciaran, I will remove `loading`. It's not needed at the moment, so we shouldn't leave it in the codes. If it's needed, it could be added later on. (Y)

ciaranschutte replied:
thanks @daniel-cy-lu !

samrichca commented:
"2023 for all copyright please. i thnk @samrichca had a script?" @ciaranschutte I had one for adding copyright to files without it i.e. in a new repo, but in this case search & replace will do

ciaranschutte commented:
> "2023 for all copyright please. i thnk @samrichca had a script?" @ciaranschutte I had one for adding copyright to files without it i.e. in a new repo, but in this case search & replace will do

for sure

ciaranschutte commented on components/pages/program-entity/ProgramCardsLayout/index.tsx:
```diff
@@ -0,0 +1,130 @@
+const ProgramCardsLayout: ComponentType<{}> = () => {
+  const theme = useTheme();
+
+  //the following three variables are mock data
+  const programSummaryData = {
+    'Program Shortname': 'PACA-CA',
+    'Full Program Name': 'Pancreatic Cancer - CA',
+    Description: 'Preventative research and treatment',
+    Countries: 'Canada',
+    'Primary Sites': 'Pancreas',
+    Website: 'https://pancreaticcancercanada.ca/',
+    Institutions: 'Pancretic Cancer Canada',
+    'Processing Regions': 'Collaboratory - Toronto, Montreal, Vancouver',
+    'Cancer Types': ' Pancreatic Cancer',
+  };
+
+  const donorAndFileCountsByDataCategory = [
+    { Category: 'Quality Control Metrics', Donors: 245, Files: 245 },
+    { Category: 'Simple Nucleotide Variation', Donors: 245, Files: 245 },
+    { Category: 'Copy Number Variation', Donors: 245, Files: 245 },
+    { Category: 'Structural Variation', Donors: 245, Files: 245 },
+    { Category: 'Transciptome Profiling', Donors: 245, Files: 245 },
+  ];
+
+  const donorAndFileCountsByExperimentalStrategy = [
+    { Strategies: 'WXS', Donors: 245, Files: 245 },
+    { Strategies: 'WGS', Donors: 245, Files: 245 },
+    { Strategies: 'RNA-Seq', Donors: 245, Files: 245 },
+  ];
```
let's put all this mock data into a separate file and import it.
easier to refactor it later.

when it comes time to remove it, we can just delete the import and have a file full of type errors instead of potential mock data spread throughout a component doing lots of work.

daniel-cy-lu replied:
Add these three variables containing mock data to a separate file, util.tsx.

ciaranschutte reviewed (changes requested):
https://github.com/icgc-argo/platform-ui/pull/2570/files#r1104961127

ciaranschutte commented:
final comment from me is the DNALoader import can be removed, because its not in use anymore

ciaranschutte commented:
on mac "shift + option + o" will cleanup and removed unused imports for you.
I usually use that hotkey after I work on a file to make sure I'm not still importing things :)

ciaranschutte commented:
i checked out the page locally and it's looking great

daniel-cy-lu commented:
> on mac "shift + option + o" will cleanup and removed unused imports for you. I usually use that hotkey after I work on a file to make sure I'm not still importing things :)

WOW super useful hot key, Ciaran!! I cleaned up all the pages I worked on in this ticket!
